### ISSUE TITLE
Detecting overriding same let in same rspec scope

### ISSUE THREAD

**Issue by Dorian** [bbatsov/rubocop#4057]
Avoid copy-paste / grammar errors by detecting overriding same rspec `let` / `let!` / `subject`, e.g.

```ruby
let(:a1) { 1 }
let(:a1) { 2 }
```

--------

## Expected behavior

This should create an offense:

```ruby
let(:a1) { 1 }
let(:a1) { 2 }
```

Correct code could be:

```ruby
let(:a1) { 1 }
let(:a2) { 2 }
```

## Actual behavior

It doesn't create an offense

## RuboCop version

```
0.47.1 (using Parser 2.3.3.1, running on ruby 2.3.3 x86_64-darwin16)
```

(mostly just a memo for myself to create this cop)

**Comment by bbatsov** [comment]
That's outside the scope of RuboCop - it should be handled by `rubocop-rspec`.

**Comment by Dorian** [comment]
@bbatsov That's so great, I didn't know about extensions! https://github.com/bbatsov/rubocop/blob/master/manual/extensions.md
